src/mahoji/lib/abstracted_commands/maniacalMonkeyCommand.ts

```diff
@@ -0,0 +1,315 @@
+			return `You need 70 Magic to use Ice Burst. You have ${user.skillLevel(SkillsEnum.Magic)}`;
+		}
+		if (user.hasEquippedOrInBank('Occult necklace')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Occult necklace');
+		}
+		if (user.hasEquippedOrInBank('Tormented bracelet')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Tormented bracelet');
+		}
+		if (user.hasEquippedOrInBank('Kodai wand')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Kodai wand');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe top')) {
+			timeToFinish *= (100 - 3) / 100;
+			boosts.push('3% for Ancestral robe top');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe bottom')) {
+			timeToFinish *= (100 - 2) / 100;
+			boosts.push('2% for Ancestral robe bottom');
+		}
+		consumableCosts.push(iceBurstConsumables);
+		timeToFinish = reduceNumByPercent(timeToFinish, 67);
+		boosts.push('67% for Ice Burst');
+	} else if (method === 'chinning' && attackStyles.includes(SkillsEnum.Ranged)) {
+		// Check what Chinchompa to use
+		const chinchompas = ['Black chinchompa', 'Red chinchompa', 'Chinchompa'];
+		let chinchompa = 'Black chinchompa';
+		for (let chin of chinchompas) {
+			if (user.owns(chin) && user.bank.amount(chin) > 5000) {
+				chinchompa = chin;
+				break;
+			}
+		}
+		const regularVoid = resolveItems([
+			'Void knight top',
+			'Void knight robe',
+			'Void knight gloves',
+			'Void ranger helm'
+		]);
+		const eliteVoid = resolveItems(['Elite void top', 'Elite void robe', 'Void knight gloves', 'Void ranger helm']);
+		if (user.hasEquippedOrInBank(eliteVoid, 'every')) {
+			timeToFinish *= (100 - 10) / 100;
+			boosts.push('10% for full Elite void equipment');
+		} else if (user.hasEquippedOrInBank(regularVoid, 'every')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Void equipment');
+		}
+		if (user.hasEquippedOrInBank('Necklace of anguish')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Necklace of anguish');
+		}
+		if (user.hasEquippedOrInBank('Twisted buckler')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Twisted buckler');
+		}
+		const chinBoostRapid = chinchompa === 'Chinchompa' ? 82 : chinchompa === 'Red chinchompa' ? 87 : 90;
+		const chinBoostLongRanged = chinchompa === 'Chinchompa' ? 77 : chinchompa === 'Red chinchompa' ? 83 : 87;
+		const chinningConsumables: Consumable = {
+			itemCost: new Bank().add(chinchompa, 1),
+			qtyPerMinute: attackStyles.includes(SkillsEnum.Defence) ? 24 : 33
+		};
+		if (attackStyles.includes(SkillsEnum.Defence)) {
+			timeToFinish = reduceNumByPercent(timeToFinish, chinBoostLongRanged);
+			boosts.push(`${chinBoostLongRanged}% for ${chinchompa}`);
+		} else {
+			timeToFinish = reduceNumByPercent(timeToFinish, chinBoostRapid);
+			boosts.push(`${chinBoostRapid}% for ${chinchompa}`);
+		}
+		consumableCosts.push(chinningConsumables);
+	}
+
+	// If no quantity provided, set it to the max.
```

Comment:
I think everything past this point is a copy and paste of minionKill with a few tweeks for activity tracking and chinning perhaps a different approach could be made.

Reply:
I made a killableMonster PR for it now, just need to fix rates when and if oldschoolJS gets updated. 
This was the approach I wanted to go with first just didn't like being locked behind another PR :(